react-toggle-switch-component/src/toggle-switch.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+class ToggleSwitch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      buttonColor: false,
+      text: false,
+      background: false
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState({
+      buttonColor: !this.state.buttonColor,
+      text: !this.state.text,
+      background: !this.state.background
+    });
+  }
+
+  render() {
+    return (
+      <div className={this.state.background ? 'div-on' : 'div-off'}>
+        <button
+          className={`btn ${this.state.buttonColor ? 'btn-on' : 'btn-off'}`}
+        onClick={this.handleClick}>
+        </button>
+        <p className='text'>{this.state.text ? 'on' : 'off'}</p>
+      </div>
+    );
```

Comment:
Since all three state values change in lock-step, please try using a _single_ state value, then deriving your three class names from that same state value.

Reply:
Updated!  sorry about the commit message it was habit to type it and hit send before I realized. Let me know if the revised code works.